fastplotlib/graphics/_base.py

```diff
@@ -93,34 +90,53 @@ def _set_feature(self, feature: str, new_data: Any, indices: Any):
         pass
 
     @abstractmethod
-    def _reset_feature(self, feature: str, old_data: Any, indices: Any):
+    def _reset_feature(self, feature: str, old_data: Any):
         pass
 
-    def link(self, event_type: str, target: Graphic, feature: str, new_data: Any, indices_mapper: callable = None):
+    def link(self, event_type: str, target: Any, feature: str, new_data: Any, indices_mapper: callable = None):
         valid_events = ["click"]
         if event_type in valid_events:
             self.world_object.add_event_handler(self.event_handler, event_type)
         else:
             raise ValueError("event not possible")
 
+        if isinstance(target.data, List):
+            old_data = list()
+            for line in target.data:
+                old_data.append(getattr(line, feature).copy())
+        else:
+            old_data = getattr(target, feature).copy()
+
         if event_type in self.registered_callbacks.keys():
             self.registered_callbacks[event_type].append(
-                CallbackData(target=target, feature=feature, new_data=new_data, old_data=getattr(target, feature).copy()))
+                CallbackData(target=target, feature=feature, new_data=new_data, old_data=old_data, indices_mapper=indices_mapper))
```

Comment:
`old_data` shouldn't be in `CallbackData`, `old_data` should be managed by the target graphic event time it handles an event. When the target graphic handles an event, it should have a copy of the feature data at the indices that are being modified and reset the feature at those indices, and then set the feature w.r.t. to the new `CallbackData`.

To elaborate:
1. The target graphic keeps a private dict of `(features, orig_vals)` that were modified at certain indices from the previous callback, something in the form `{"feature": (indices, orig_vals_at_indices)}`, you could call it `self._previous_modified`. And with the new `GraphicFeature` class from #78 we won't even need a dict for it, we can just add `_modified_indices` and `_orig_vals_at_modified_indices` as private attributes to `GraphicFeature` :smile: .

2. In `_set_feature()` it first uses the information from above to reset the feature at those indices to their original values

Reply:
with #78 it might be possible for us to just pass the GraphicFeature instance _directly_ instead of the string name when calling `link`, for example:

`graphic.link(event_type, target=some_graphic, feature=some_graphic.color...`

Reply:
If info for resetting feature is moved to ```_set_feature()``` graphic, then we will no longer need method for ```_reset_feature()```?

Reply:
we should still have a `_reset_feature()` to separate functionality